oracledb-ballerina/client.bal

```diff
@@ -0,0 +1,213 @@
+
+import ballerina/crypto;
+import ballerina/jballerina.java;
+import ballerina/sql;
+
+public client class Client {
+    *sql:Client;
+    private boolean clientActive = true;
+
+    # Initialize Oracle Client.
+    #
+    # + user - Name of a user of the database
+    # + password - Password for the user
+    # + database - System Identifier or the Service Name of the database
+    # + host - Hostname of the oracle server to be connected
+    # + port - Port number of the oracle server to be connected
+    # + options - Oracle database specific JDBC options
+    # + connectionPool - The `sql:ConnectionPool` object to be used within 
+    #         the jdbc client. If there is no connectionPool provided, 
+    #         the global connection pool will be used
+    public isolated function init(
+        string user,
+        string password,
+        string host = "localhost",
+        int port = 1521,
+        string database = "ORCL",
+        Options? options = (),
+        sql:ConnectionPool?  connectionPool = ()
+    ) returns sql:Error? {
+        ClientConfiguration clientConfig = {
+            user: user,
+            password: password,
+            host: host,
+            port: port,
+            database: database,
+            options: options,
+            connectionPool: connectionPool
+        };
+        return createClient(self, clientConfig, sql:getGlobalConnectionPool());
+    }
+
+    # Queries the database with the query provided by the user, and returns the result as stream.
+    #
+    # + sqlQuery - The query which needs to be executed as `string` or `ParameterizedQuery` when the SQL query has
+    #              params to be passed in
+    # + rowType - The `typedesc` of the record that should be returned as a result. If this is not provided the default
+    #             column names of the query result set be used for the record attributes.
+    # + return - Stream of records in the type of `rowType`
+    remote isolated function query(@untainted string|sql:ParameterizedQuery sqlQuery, typedesc<record {}>? rowType = ())
+    returns @tainted stream <record {}, sql:Error> {
+        if (self.clientActive) {
+            return nativeQuery(self, sqlQuery, rowType);
+        } else {
+            return sql:generateApplicationErrorStream("OracleDB client is already closed,"
+                + "hence further operations are not allowed");
+        }
+    }
+
+    # Executes the DDL or DML sql queries provided by the user, and returns summary of the execution.
+    #
+    # + sqlQuery - The DDL or DML query such as INSERT, DELETE, UPDATE, etc as `string` or `ParameterizedQuery`
+    #              when the query has params to be passed in
+    # + return - Summary of the sql update query as `ExecutionResult` or returns `Error`
+    #           if any error occurred when executing the query
+    remote isolated function execute(@untainted string|sql:ParameterizedQuery sqlQuery) returns sql:ExecutionResult|sql:Error {
```

Comment:
Adhere to the best practice of max line length of 120 chars. Check all the places.

Reply:
Fixed with 28468e1